lib/iris/fileformats/grib/_message.py

```diff
@@ -110,6 +108,8 @@ def data(self):
                 raise TranslationError(msg)
             if template == 90:
                 shape = (grid_section['Ny'], grid_section['Nx'])
+            elif template == 40 and reduced:
+                shape = (grid_section['numberOfDataPoints'],)
```

Comment:
If `reduced` is only supported for template 40 then shouldn't we still be raising an error if it occurs with other templates? Or is that just not possible? (I don't have the spec to hand at the moment.)